### ISSUE TITLE
Error SSL

### ISSUE THREAD

**Issue by Romounet** [GoSecure/pyrdp#78]
Sorry I could not reproduce this:

```[2019-02-27 13:11:23,656] - mitm                                               - INFO     - MITM Server listening on port 3389
[2019-02-27 13:11:37,521] - mitm.connections.Ella293529.server                 - INFO     - TCP connected from 192.168.160.151:51724
[2019-02-27 13:11:37,522] - mitm.connections.Ella293529.server                 - INFO     - No cookie for this connection 

Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/Twisted-18.9.0-py3.6-linux-x86_64.egg/twisted/python/log.py", line 103, in callWithLogger
    return callWithContext({"system": lp}, func, *args, **kw)
  File "/usr/local/lib/python3.6/dist-packages/Twisted-18.9.0-py3.6-linux-x86_64.egg/twisted/python/log.py", line 86, in callWithContext
    return context.call({ILogContext: newCtx}, func, *args, **kw)
  File "/usr/local/lib/python3.6/dist-packages/Twisted-18.9.0-py3.6-linux-x86_64.egg/twisted/python/context.py", line 122, in callWithContext
    return self.currentContext().callWithContext(ctx, func, *args, **kw)
  File "/usr/local/lib/python3.6/dist-packages/Twisted-18.9.0-py3.6-linux-x86_64.egg/twisted/python/context.py", line 85, in callWithContext
    return func(*args,**kw)
--- <exception caught here> ---
  File "/usr/local/lib/python3.6/dist-packages/Twisted-18.9.0-py3.6-linux-x86_64.egg/twisted/internet/asyncioreactor.py", line 136, in _readOrWrite
    why = method()
  File "/usr/local/lib/python3.6/dist-packages/Twisted-18.9.0-py3.6-linux-x86_64.egg/twisted/internet/tcp.py", line 243, in doRead
    return self._dataReceived(data)
  File "/usr/local/lib/python3.6/dist-packages/Twisted-18.9.0-py3.6-linux-x86_64.egg/twisted/internet/tcp.py", line 249, in _dataReceived
    rval = self.protocol.dataReceived(data)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/tcp.py", line 81, in dataReceived
    self.next.recv(data)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/segmentation.py", line 88, in recv
    layer.recv(forwarded)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/buffered.py", line 52, in recv
    self.pduReceived(pdu, self.hasNext)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/layer.py", line 126, in pduReceived
    self.next.recv(pdu.payload)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/x224.py", line 79, in recv
    self.pduReceived(pdu, pdu.header == X224PDUType.X224_TPDU_DATA)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/layer.py", line 123, in pduReceived
    self.observer.onPDUReceived(pdu)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/core/observer.py", line 84, in __call__
    self.composite.doCall(self.item, args, kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/core/observer.py", line 56, in doCall
    getattr(observer, item)(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/layer/layer.py", line 48, in onPDUReceived
    self.handlers[pdu.header](pdu)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/mitm/client.py", line 148, in onConnectionConfirm
    self.server.onConnectionConfirm(pdu)
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/mitm/server.py", line 238, in onConnectionConfirm
    certificateFileName=self.certificateFileName))
  File "/usr/local/lib/python3.6/dist-packages/pyrdp-1.0.0-py3.6-linux-x86_64.egg/pyrdp/core/ssl.py", line 51, in __init__
    TPDUSSLContext)
  File "/usr/local/lib/python3.6/dist-packages/Twisted-18.9.0-py3.6-linux-x86_64.egg/twisted/internet/ssl.py", line 108, in __init__
    self.cacheContext()
  File "/usr/local/lib/python3.6/dist-packages/Twisted-18.9.0-py3.6-linux-x86_64.egg/twisted/internet/ssl.py", line 117, in cacheContext
    ctx.use_certificate_file(self.certificateFileName)
  File "/usr/lib/python3/dist-packages/OpenSSL/SSL.py", line 935, in use_certificate_file
    _raise_current_error()
  File "/usr/lib/python3/dist-packages/OpenSSL/_util.py", line 54, in exception_from_error_queue
    raise exception_type(errors)
OpenSSL.SSL.Error: [('SSL routines', 'SSL_CTX_use_certificate', 'ca md too weak')]

[2019-02-27 13:11:37,551] - mitm.connections.Ella293529.server                 - INFO     - Connection closed: [Failure instance: Traceback (failure with no frames): <class 'twisted.internet.error.ConnectionAborted'>: Connection was aborted locally, using.

```

**Comment by Res260** [comment]
https://forums.openvpn.net/viewtopic.php?t=23979

Who uses MD5 to sign certificates 🤔 

This is a problem related to OpenSSL, I don't think there is a way to loosen that restriction, but I might be wrong.

**Comment by xshill** [comment]
@Romounet It looks like the problem was that the certificate _you_ were trying to use was signed with MD5. Did you use the certificate cloner to generate it? If so, it might be good to warn the user that MD5 is used and the certificate might not work.

Like @Res260 said, this is an OpenSSL thing and I don't think there's really a good way to prevent it from rejecting MD5.

**Comment by xshill** [comment]
Closed by c9a479d806a4c65d174f95d49bb9cca25628450d and ffe0c5b7b8b4731f3e32616b505e4bb198028190.
